solidity/contracts/Oracle.sol

```diff
@@ -1,14 +1,13 @@
 pragma solidity ^0.4.24;
 
-import "./interfaces/OracleInterface.sol";
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
-import "link_token/contracts/LinkToken.sol";
+import "./interfaces/LinkTokenInterface.sol";
 
-contract Oracle is OracleInterface, Ownable {
+contract Oracle is Ownable {
   using SafeMath for uint256;
 
-  LinkToken internal LINK;
+  LinkTokenInterface internal LINK;
```

Comment:
There's a problem with using the LinkTokenInterface in the Oracle contract. If we use the interface here, Truffle never generates the artifacts for the actual LinkToken contract. The migration script `solidity/migrations/2_link_token_contract.js` requires the real LinkToken contract's artifacts to be present in order to deploy.

When I try to run `./internal/ci/truffle_test` locally, I receive the following error:

```
Error: Could not find artifacts for ./LinkToken from any sources
    at Resolver.require (/home/thomas/go/src/github.com/smartcontractkit/chainlink/node_modules/truffle/build/cli.bundled.js:59973:9)
    at TestResolver.require (/home/thomas/go/src/github.com/smartcontractkit/chainlink/node_modules/truffle/build/cli.bundled.js:95332:30)
    at Object.require (/home/thomas/go/src/github.com/smartcontractkit/chainlink/node_modules/truffle/build/cli.bundled.js:69602:36)
    at ResolverIntercept.require (/home/thomas/go/src/github.com/smartcontractkit/chainlink/node_modules/truffle/build/cli.bundled.js:197047:32)
    at /home/thomas/go/src/github.com/smartcontractkit/chainlink/solidity/migrations/2_link_token_contract.js:1:27
    at ContextifyScript.Script.runInContext (vm.js:59:29)
    at ContextifyScript.Script.runInNewContext (vm.js:65:15)
    at /home/thomas/go/src/github.com/smartcontractkit/chainlink/node_modules/truffle/build/cli.bundled.js:101639:14
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3)
```

Even if you specified the path to `link_token/contracts/LinkToken.sol`, Truffle still wouldn't generate artifacts for it.

Reply:
Wondering why do we need migrations? Do we still use them anymore?